features/step_definitions/report_steps.rb

```diff
@@ -1,9 +1,18 @@
-Given /^test result report (.+) was generated(?: with UID ([a-zA-Z0-9_-]+))? for organization (.+)$/ do |name,uid,organization_name|
+Given /^test result report (.+) was generated(?: (\d+) day(?:s)? ago)?(?: with UID ([a-zA-Z0-9_-]+))? for organization (.+)$/ do |name,days,uid,organization_name|
   options = {
     uid: uid,
     organization: named_record(organization_name)
   }
 
+  if days
+    date = days.to_i.days.ago
+
+    options.merge!({
+      started_at: date - 3.minutes,
```

Comment:
I would use `date` as is for `started_at` and add some time for `ended_at` instead of subtracting for both. Agreed, it's unlikely, but technically this could end up not being the same start & end day, whereas if you only add to `ended_at`, at least the start date will always correspond to what is written in the scenario.